reporters/dogstatsd.go

```diff
@@ -84,3 +111,23 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 func NewDogStatsDFromClient(c dogstatsd.Client, r string) *DogStatsD {
 	return &DogStatsD{client: c, region: r}
 }
+
+func (d *DogStatsD) restartTicker() {
+	for range d.ticker.C {
+		d.mutex.Lock()
+		err := d.statsdClient.Close()
+		d.mutex.Unlock()
```

Comment:
Oh, did you change this back? It was correct before. You should have an Unlock before every end of the loop. Be a `continue`, be the end of the bracket. 

Maybe my comment was misleading? What I meant was this:
1. Have a submethod, such as (with a few changes too):
```
func (d *DogStatsD) restartTicker() error {
		err := d.statsdClient.Close()
		if err != nil {
			return fmt.Errorf("failed to close statsd connection: %w", err)
		}

		c, err := dogstatsd.New(d.host, d.prefix)
		if err != nil {
			return fmt.Errorf("failed to recreate dogstatsd client:%w", err)
		}

		d.statsdClient = c.Client.(*statsd.Client)
		d.client = c
}
```

2. Call the submethod in the loop method:

```
func (d *DogStatsD) restartTickerInLoop() {
		for range d.ticker.C {
				d.mutex.Lock()
				if err := d.restartTicker(); err != nil {
						d.logger.Errorf("DogStatsD: failed to close statsd connection during restart: %s", err.Error())
				}
				d.mutex.Unlock()
		}
}
```

But, please, I was ok with the previous solution. You can also just revert it back. As long as the Lock/Unlock encompass the whole close+assignment.

Reply:
Oh! 🤦🤦🤦 Sorry, I misunderstood.

I applied your suggestion. I agree with you. It's more clear and can avoid misinterpretation in the future.

Sorry again for taking your time on this review 🙇‍♂️

Reply:
Hey, Tobias, don't worry about it 🙂